src/ChaosBox/CLI.hs

```diff
@@ -265,15 +277,55 @@ runChaosBoxDirectly Opts {..} doRender = replicateM_ optRenderTimes $ do
       , gcMetadataString = optMetadataString
       }
 
-  void . renderWith surface . flip runReaderT ctx . flip runRandT stdGen $ do
-    cairo $ scale optScale optScale
-    void doRender
+  if optFileFormat == PNG || optRenderMode == Interactive then
+    void . renderWith surface . flip runReaderT ctx . flip runRandT stdGen $ do
+      cairo $ scale optScale optScale
+      void doRender
+      
+      ref   <- asks gcBeforeSaveHook
+      mHook <- liftIO $ readIORef ref
+      fromMaybe (pure ()) mHook
 
-    ref   <- asks gcBeforeSaveHook
-    mHook <- liftIO $ readIORef ref
-    fromMaybe (pure ()) mHook
+      saveImage
 
-    saveImage
+  else case optFileFormat of
+    SVG -> write ctx stdGen (fromIntegral w) (fromIntegral h) withSVGSurface
+    PS  -> write ctx stdGen (fromIntegral w) (fromIntegral h) withPSSurface
+    PDF -> write ctx stdGen (fromIntegral w) (fromIntegral h) withPDFSurface
+    PNG -> error "impossible"
```

Comment:
To avoid this, you could case on both `optFileFormat` and `optRenderMode` here:

```haskell
case (optFileFormat, optRenderMode) of
  (PNG, Interactive) ->  {- lines 281 - 289 above -}
  (PNG, Static) -> {- write png -}
  (PS, Static) -> {- write ps -}
  (PDF, Static) -> {- write pdf -}
  (SVG, Static) -> {- write svg -}
  (fileFormat, renderMode) ->
    error $ show renderMode <> " mode does not support rendering to " <> show fileFormat
```

Reply:
That's much nicer indeed; I changed it.